readme.txt
!!!REQUIRES COUNTER-STRIKE:SOURCE!!!
!!!RECOMMENDED MAXIMUM AMOUNT OF PLAYERS FOR A SINGLE SERVER IS 12-16!!!

This gamemode is similar to TTT in terms of the gamemode type, but is different in almost every aspect.

TL;DR: TTT but much more focused on espionage and is not a massacre simulator.

Long version:

The first thing you'll notice is that the gamemode is not a town murder simulator.
Out of all players 3 are chosen, 2 of which are Agent A and Agent B, and the 3rd person is the VIP.
Everyone else will get their role set to Civilian, who are sworn to protect the VIP.

Agents have to kill each other AND the VIP to win the round, while the VIP has to either kill the agents or wait for the round to end.
Each player is able to examine players, which, if he succeeds, will notify the player that the examined person is trusted/suspicious/dangerous.
Agents are great at examining players and have a boost in examining people, everyone else do not have this boost, but can buy a "manual" from the
buy menu and get the boost.

How it works: Every weapon you carry and body armor you have equipped increases your "weight" value and when it exceeds a certain number, then upon examination
the player will be notified that you might have something concealed.

Agents are also much more experienced with firearms than the VIP and the Civilians are. (less recoil, less spread)

Upon round start, the VIP gets a VIP ID card, which he can show to people to ensure them that he is the VIP and have them as his bodyguards.
Agents can buy forged VIP ID cards, which they can use the same way the VIP does, but to cause confusion and to make people believe that he is
the VIP. However, the forgery can be noticed by an opposing Agent and the VIP, which will automatically mark the impostor as an Agent on their
side.

There's also a bunch more shit that's different from TTT and it would take me a wall of text to write all the differences and shit like that.

Features:
- Random player selection for Agents/VIPs (essential for a gamemode like this)
- Round system (same as above)
- Various ways of tricking people as an Agent
- Money system (nothing special, really)
- Agents can pick-pocket people to find clues
- Buyable hints for Agents (a la Hitman: Blood Money)
- Uses my "Customizable Weaponry" (without the attachments part) for the firearms
- 3 armor vest types that protect only the torso of a player
- Karma system (similar to TTT, I think?)
- Dead body examination (similar to TTT)
- Round system
- Map voting system
- Pain sounds
- Limited, slow health regen
- Mail system for supported maps
- Stamina system
- Drowning (that's right, players can drown!)
- Low ledge-grabbing system to help overcome low obstacles (such as walls)
- Players can unload weapons that are lying on the floor
- Tips when dead
- Ability to break glass without having to use a weapon
- Push around props and play spooky sounds as a spectator to keep yourself entertained
- Buyable radios which allow for long-range communication and the use of a voice chat menu

HOW TO INSTALL THE GAMEMODE CONTENT PROPERLY:
Copy-paste the contents of the folder 'content' into your server's root folder (where 'materials', 'models', 'sound' folders are)
The gamemode is then installed like any other gamemode.

Here are the maps that support the gamemode:
cs_assault
cs_compound
cs_italy
cs_havana
cs_militia
cs_office
de_cbble
de_inferno
de_nuke
de_chateau
de_piranesi
de_port
de_tides

Any map that's NOT supported, will not feature the mailbox system and instead will make item buys instant.
Be sure to add them and ONLY THEM into your server's rotation, as that's where the maps for the votemap are taken from!
You can just copy-paste these maps to mapcycle.txt and it should work.

Server-side console variables:
dec_roundtime <default - 600> - time in seconds until the round ends
dec_runspeed <default - 310> - run speed for every player
dec_startmoney_civilian <default - 200> - the amount of money a civilian starts out with
dec_startmoney_agent <default - 350> - the amount of money an Agent starts out with
dec_startmoney_vip <default - 500> - the amount of money a VIP starts out with
dec_income_civ_time <default - 2> - the time it takes for a Civilian to receive the money
dec_income_civ_amount <default - 1> - the amount of money a Civilian receives
dec_income_agent_time <default - 1> - the time it takes for an Agent to receive the money
dec_income_agent_amount <default - 1> - the amount of money an Agent receives
dec_income_vip_time <default - 1> - the time it takes for the VIP to receive the money
dec_income_vip_amount <default - 2> - the amount of money the VIP receives
dec_healthregen_enabled <default - 1> - disable or enable health regen
dec_healthregen_percentage <default - 0.25> - percentage of received damage that the player won't regenerate
dec_healthregen_dmgpenalty <default - 15> - how long does the player have to wait for the health regen to start after taking damage (in seconds)
dec_healthregen_regentime <default - 2> - how long does the player have to wait for the health to regenerate (in seconds)
dec_stamina_drain_sprint_time <default - 0.6> - amount of time it takes to decrease stamina while sprinting (in seconds, applies to swimming fast too)
dec_stamina_drain_sprint_amount <default - 1> - amount of stamina you lose while sprinting (applies to swimming fast too)
dec_stamina_drain_ledgepull <default - 10> - amount of stamina you lose every time you pull yourself up
dec_stamina_drain_ledgehang_time <default - 1> - amount of time it takes to decrease stamina while hanging on a ledge (in seconds)
dec_stamina_drain_ledgehang_amount <default - 1> - amount of stamina you lose while hanging on a ledge
dec_stamina_drain_jump <default - 5> - amount of stamina you lose every time you jump
dec_stamina_drain_underwater_time <default - 0.3> - amount of time it takes to decrease stamina while underwater
dec_stamina_drain_underwater_amount <default - 1> - amount of stamina you lose when underwater
dec_stamina_regen_time <default - 0.2> - the delay between stamina regen
dec_stamina_regen_amount <default - 1> - amount of stamina to regen
dec_pickup_enabled <default - 1> - allow or disallow the ability to pick props up
dec_pickup_weight <default - 30> - the maximum weight of a prop that a player can pick up. If the prop's weight exceeds this value, the player won't be able to pick it up.
dec_chat_distance <default - 256> - the maximum distance a player can see another player use his chat
dec_mic_distance <default - 256> - the maximum distance a player can hear another player use his microphone
dec_rounds <default - 7> - amount of rounds per map
dec_max_armedkills <default - 3> - after killing this many armed Civilians, the Agent will begin to lose karma
dec_bandur <default - 60> - time in minutes that a player gets banned for having a karma that's lower than 60%
dec_noreturnfirekarmaloss <default - 1> - disable or enable karma loss for killing armed Civilians that did not return fire as an Agent
dec_karmalossmaxdamage <default - 175> - if a Civilian does more damage than this to other Civilians, then he can be killed by other Civilians without fear of losing karma
dec_confirm_agent <default - 100> - amount of money an Agent gets for confirming a dead body of an Agent
dec_confirm_vip <default - 200> - amount of money an Agent gets for confirming a dead body of the VIP
dec_maxweapons <default - 2> - amount of weapons a player can carry
dec_maxweapons_unique <default - 2> - amount of unique weapons a player can carry

Admin-only console commands:
dec_restartround - restarts the round

FAQ:

Q: I have been RDM'ed!
A: There is no such thing as RDM in this gamemode. If you have been killed it means that you were suspected of being an Agent/VIP. Stop whining.

Q: What does karma do?
A: Karma increases or decreases starting money. If your karma goes below 60%, you get kicked.

Q: I am moving my mouse wheel, but the amount of money I should give to a player does not increase.
A: The mouse wheel has to be bound to next/previous weapon selection in order to work.

Credits:
trotskygrad - ideas, HUD concept
Kogitsune - ideas
Sam6420 - ideas
Jvs - tips on how to optimize client-side stuff

P.S. I have created a thread for this gamemode on Facepunch Forums (www.facepunch.com). If you find a bug or have a suggestion, look up the thread there, and post your bug/suggestion.

Known bugs:
Crouching down with a concealed knife makes you look like you have it drawn - no idea how to fix this

Changelog.txt
1.0.6.2:
	/\ Fixed the Ballistics Analyzer breaking graphics
	/\ Fixed a client-side bug that sometimes did not reset players' statuses upon round end

1.0.6.1:
	+ Added a console command to restart the round (UNDOCUMENTED IN 1.0.6)
	/\ If there are 3 players on, but one of them is dead, then the round won't start until there are 3 living people
	/\ Dead players will not be chosen as Agents/VIP when a round starts
	/\ Fixed a small client-side bug

1.0.6:
	New gamemode content added - please, copy-paste the contents of the "content" folder into your server's root folder!
	
	+ Added a new weapon/item - Ballistics Analyzer:
		* It can be used to find out the killer by sampling the bullet on dead bodies and then comparing the fingerprints on it to other players' fingerprints
		* The percentage of analysis increase per body find examination and the initial amount of percentage can be changed via console variables
		* It is a unique weapon
		
	+ Added a console variable to control the maximum amount of weapons a player can carry
	+ Added a console variable to control the maximum amount of unique weapons a player can carry
	+ Added on-screen icons in the upper left corner of the screen for items such as the Radio, Antidote, etc
	+ Added an admin-only restart round command
	+ Added 3 more tips
	
	/\ Improved the Walkie-talkie menu:
		* Added 1 new category (which lets me add more responses)
		* Added 3 new response options
		* Added more sounds to several responses
		* Fixed some bugs that unaffected anything
		
	/\ The models of the UMP .45, MP5 and P90 are now visible on players' backs when they are holstered
	/\ Now when an Agent or the VIP leaves the server, every player will get notified
	/\ Unique weapons now do not increase the maximum weapon limit (such as Knife, Dart Gun)
	/\ Changed "Trusted" to "Unarmed" and "Dangerous" to "Armed"
	/\ Fixed Civilians attacking other Civilians resulting in the victim's "damage done" increasing, instead of the attacker (what)
	/\ The Antidote upon purchase now gets stored in the player's inventory. You can now use it by pressing 6
	/\ The icons of items such as Antidotes, Walkie-talkies are now displayed in the left upper corner
	/\ The Dart Gun now drains 100 health points in 8 seconds (was 10 seconds)
	/\ Corrected some weapons' description text being incorrect
	/\ Increased Antidote price to 25$ (was 10$)
	
1.0.5.2:
	/\ Fixed Civilians doing more than X amount of damage to Civilians not decreasing their karma

1.0.5.1:
	/\ "How to examine quickly" now increases exam speed by 1 sec instead of 0.75 sec
	/\ Fixed not being able to select a different weapon using the mouse wheel if the player looked at a player some time ago
	/\ Fixed a spam of instructions when you become an Agent at the start of a round

1.0.5:
	+ Players now have their sprint speed slowed down to 50% after taking damage (the slow-down dissapears in about 1-2 secs)
	+ When a player has 20 health or less, his sprint speed increase will be 15% (as opposed to 138% when healthy) and he won't be able to grab onto ledges to pull himself up
	+ Now if a Civilian does more than 175 damage points to Civilians/VIP, then that player can be killable with no karma loss as a Civilian (the amount of damage is controllable via convar)
	+ Added 14 more tips
	+ Players can now give each other money - to control the amount of money to give use your mouse wheel (next and previous weapon binds) and press your noclip key (default - V) to give the amount
	+ Now if an Agent examines a dead body and there are players nearby, then he will report the dead body
	+ Added a "Report body" button for Agents when they're examining dead bodies and need to report dead bodies in order to prevent giving themselves away
	+ The first Agent who identifies the dead body of an Agent or the VIP will receive money. (100$ for an Agent, 200$ for the VIP, can be changed via console variables)
	/\ Holding down your sprint key while using the voice chat now increases it's range 2x (you pretty much shout this way, which is useful when there are lots of people running somewhere and you need to notify them of something)
	/\ Joining a different radio channel now wipes the current master password information
	/\ Joining the default radio channel now sets the correct channel info
	/\ Sprinting left/right/back is now slower than sprinting forward
	/\ Server admins can now see the names of people who use voice chat and regular chat (for administrative purposes)
	/\ Fixed armor not working
	/\ Decreased maximum carryable poison darts from 10 to 5
	
	/\ Changed the stamina loss when receiving damage:
		* If you get hit in the arms/torso/etc. you lose 50% stamina based on the damage
		* If you get hit in the head, you lose 200% stamina based on the damage
		* If you get hit in the legs, you lose 125% stamina based on the damage
		* Example: you receive 20 damage, arms/torso = 10 stamina; legs = 25 stamina; head = 40 stamina
		
	/\ Taking damage now also delays stamina regen by one second
	/\ Fixed using the team chat without a Walkie-talkie resulting in global chat (now only you can see your message if you use team chat without a Walkie-talkie)
	/\ Fixed players typing in the chat displaying their names, even if the player wasn't examined
	/\ Fixed the next tip key displaying as not bound only if your spawn menu key was not bound
	/\ Changed (TEAM) text to (RADIO) when using the team chat with a Walkie-talkie
	/\ If dec_noreturnfirekarmaloss is enabled, then players will now lose 2 instead of 10 karma points
	/\ Slight client-side optimization
	/\ dec_noreturnfirekarmaloss now defaults to 0. Note: if you started the gamemode once already, it will not default to 0

1.0.4.1:
	Sorry about this update, folks, but some people disliked one thing, so I added a convar
	+ Added a convar to enable/disable karma loss for killing armed Civilians that did not return fire as an Agent 

1.0.4:

gamemode/cl_menus.lua
local clr, rect, orect, trect, stext, create = surface.SetDrawColor, surface.DrawRect, surface.DrawOutlinedRect, surface.DrawTexturedRect, draw.SimpleText, vgui.Create
local CreatedPanels, BuyPanel, FakePanel, Cats, PSheet, MainFrame, MainFrame2, MainFrame3, MainFrame4, MainFrame5, amt, ModelPanel, RepButt, Name, Desc, Stats, ply, Stop, body, cal, time, role, ply, Wide, Tall, W, x, y, w, h, sl, st, sr, sb, pl, pt, pr, pb, dosh, PrevMins, PrevMaxs, KillCaliber, KillTime, KillRole, min, text, rolet, HelpPanel, Help, mblur, stamblur, stambeat, CBox, addy
local DFrame2 = {}

function DFrame2:Init()
end

function DFrame2:Paint()
	Wide = self:GetWide()
	Tall = self:GetTall()

	clr(137, 165, 126, 255)
	orect(0, 0, Wide, Tall)

	clr(40, 80, 40, (self.Alpha or 100))
	rect(1, 1, Wide - 2, Tall - 2)

	draw.RoundedBox(8, 3, 3, Wide - 6, 20, Color(50, 50, 50, 100))
end

vgui.Register("DFrame2", DFrame2, "DFrame")

local DPropertySheet2 = {}

function DPropertySheet2:Init()
end

function DPropertySheet2:Paint()
	Wide = self:GetWide()
	Tall = self:GetTall()

	clr(137, 165, 126, 255)
	orect(0, 22, Wide, Tall - 22)

	clr(0, 0, 0, 230)
	rect(1, 23, Wide - 2, Tall - 24)
end

vgui.Register("DPropertySheet2", DPropertySheet2, "DPropertySheet")

local Stats = {}

function Stats:SetStats(dmg, rof, acc, rng)
	self.Damage = dmg
	self.Rof = rof
	self.Accuracy = acc
	self.Range = rng
end

function Stats:Paint()
	if self.Damage and self.Rof and self.Accuracy and self.Range then
		-- optimization, bitch nigga

		clr(137, 165, 126, 255)
		orect(50, 0, 100, 12)
		orect(50, 14, 100, 12)
		orect(50, 28, 100, 12)
		orect(50, 42, 100, 12)

		clr(40, 40, 40, 255)
		rect(51, 1, 98, 10)
		rect(51, 15, 98, 10)
		rect(51, 29, 98, 10)
		rect(51, 43, 98, 10)

		stext("Damage", "DefaultSmall", 0, 0, Color(255, 255, 255, 255))
		stext("Speed", "DefaultSmall", 0, 14, Color(255, 255, 255, 255))
		stext("Accuracy", "DefaultSmall", 0, 28, Color(255, 255, 255, 255))
		stext("Range", "DefaultSmall", 0, 42, Color(255, 255, 255, 255))

		clr(137, 165, 126, 255)
		rect(52, 2, self.Damage * 0.96, 8)
		rect(52, 16, self.Rof * 0.96, 8)
		rect(52, 30, self.Accuracy * 0.96, 8)
		rect(52, 44, self.Range * 0.96, 8)
	end
end

vgui.Register("Stats", Stats, "DPanel")

local PANEL = {}

function PANEL:Paint(...)
	W = self:GetWide()
	clr(70, 100, 70, 150)
	rect(1, 1, W - 2, self:GetTall() - 2)

    x, y = self:LocalToScreen( 0, 0 )
    w, h = self:GetSize()

    sl, st, sr, sb = x, y, x + w, y + h

    p = self

    while p:GetParent() do
        p = p:GetParent()
        pl, pt = p:LocalToScreen( 0, 0 )
        pr, pb = pl + p:GetWide(), pt + p:GetTall()
        sl = sl < pl and pl or sl
        st = st < pt and pt or st
        sr = sr > pr and pr or sr
        sb = sb > pb and pb or sb
    end

    render.SetScissorRect( sl, st, sr, sb, true )
        self.BaseClass.Paint( self, ... )
    render.SetScissorRect( 0, 0, 0, 0, false )

	clr(137, 165, 126, 255)
	orect(0, 0, W, 64)
end

function PANEL:PaintOver()
	if self.CanDraw then
		dosh = LocalPlayer():GetDTInt(0)

		if dosh >= self.Price then
			draw.ShadowedText("Purchase", "TabLarge", 32, 32, Color(255, 255, 255, 255), Color(0, 0, 0, 255), 1, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER)
		else
			draw.ShadowedText("Need " .. self.Price - dosh .. "$", "TabLarge", 32, 32, Color(255, 0, 0, 255), Color(0, 0, 0, 255), 1, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER)
		end
	end
end

-- enum 107 is left click
-- enum 108 is right click

function PANEL:OnMousePressed(enum)
	ply = LocalPlayer()

	if not ply:Alive() then
		return
	end

	ply:ConCommand("dec_buyitem " .. self.ItemNum)
end

function PANEL:OnCursorEntered(enum)
	self.CanDraw = true
end

function PANEL:OnCursorExited(enum)
	self.CanDraw = false
end

vgui.Register( "DModelPanel2", PANEL, "DModelPanel" )

function PANEL:CenterView()
	PrevMins, PrevMaxs = self.Entity:GetRenderBounds()
	self:SetCamPos(PrevMins:Distance(PrevMaxs)*Vector(0.75, 0.75, 0.5))
	self:SetLookAt((PrevMaxs + PrevMins) / 2)
end

function PANEL:LayoutEntity()
end

PANEL = {}

function PANEL:Paint()
	return false
end

vgui.Register("DPanelList2", PANEL, "DPanelList")

local DImage2 = {}

function DImage2:Paint()
	if self.Texture and self.X and self.Y then
		if not self.FinalTexture then
			self.FinalTexture = surface.GetTextureID(self.Texture)
		else
			surface.SetTexture(self.FinalTexture)
			clr(255, 255, 255, 255)
			trect(0, 0, self.X, self.Y)
		end
	end
end

vgui.Register("DImage2", DImage2, "DImage")

local CanContinue = false

local function DEC_BuyMenu()
	if not RoundTime or RoundTime == -1 then
		GAMEMODE:AddNotify("Cannot purchase items until round start!", NOTIFY_ERROR, 7)
		surface.PlaySound("buttons/button10.wav")
		return
	end

	ply = LocalPlayer()

	if not ply:Alive() then
		GAMEMODE:AddNotify("Cannot purchase items while dead!", NOTIFY_ERROR, 7)
		surface.PlaySound("buttons/button10.wav")
		return
	end

	MainFrame4 = create("DFrame2")
	MainFrame4:SetSize(400, 500)
	MainFrame4:Center()
	MainFrame4:SetTitle("Buy menu")
	MainFrame4:SetVisible(true)
	MainFrame4:SetDraggable(false)
	MainFrame4:ShowCloseButton(true)
	MainFrame4:MakePopup()

	PSheet = create("DPropertySheet2", MainFrame4)
	PSheet:SetPos(25, 25)
	PSheet:SetSize(MainFrame4:GetWide() - 50, MainFrame4:GetTall() - 40)

	CreatedPanels = {}
	BuyPanel = nil
	FakePanel = nil
	Cats = {}
	CanContinue = false

	for k, v in pairs(Items) do
		for k2, v2 in pairs(v.TeamText) do
			if ply.TeamText == v2 or (not ply.TeamText and v2 == "Civilian") then
				CanContinue = true
				Stop = false
			end

			if CanContinue and not Stop then
				if not CreatedPanels[v.Category] then
					CreatedPanels[v.Category] = {}

					BuyPanel = create("DPanelList", PSheet)
					BuyPanel:SetPos(10, 20)
					BuyPanel:SetSize(200, 600)
					BuyPanel:SetSpacing(8)
					BuyPanel:EnableHorizontal(false)
					BuyPanel:EnableVerticalScrollbar(true)
					BuyPanel.Paint =  function() end

					FakePanel = create("Panel", BuyPanel)

					PSheet:AddSheet(v.Category, BuyPanel, "deception/money", false, false, v.Category)

					Cats[v.Category] = {amt = 0}
				end

				amt = Cats[v.Category].amt

				ModelPanel = create("DModelPanel2", FakePanel)
				ModelPanel:SetModel(v.Model)
				ModelPanel:SetSize(64, 64)
				ModelPanel:SetPos(10, 24 + amt * 120)
				ModelPanel:CenterView()
				ModelPanel.Price = v.Price
				ModelPanel.ItemNum = v.num

				Name = create("DLabel", FakePanel)
				Name:SetPos(12, 8 + amt * 120)
				Name:SetText(v.Name .. (v.Unique and " (U)" or "").. " - " .. v.Price .. "$")
				Name:SetFont("TabLarge")
				Name:SizeToContents()

				if v.Stats then
					Desc = create("DLabel", FakePanel)
					Desc:SetPos(12, 90 + amt * 120)
					Desc:SetText(v.Description)
					Desc:SizeToContents()

					Stats = create("Stats", FakePanel)
					Stats:SetSize(250, 60)
					Stats:SetPos(80, 28 + amt * 120)
					Stats:SetStats(v.Stats.damage, v.Stats.rof, v.Stats.accuracy, v.Stats.range)
				else
					Desc = create("DLabel", FakePanel)
					Desc:SetPos(80, 28 + amt * 120)
					Desc:SetText(v.Description)
					Desc:SetTextColor(Color(175, 255, 175, 255))
					Desc:SizeToContents()
				end

				Cats[v.Category].amt = Cats[v.Category].amt + 1
				FakePanel:SetSize(400, Cats[v.Category].amt * 120)

				if not CreatedPanels[v.Category].Registered then
					BuyPanel:AddItem(FakePanel)
					CreatedPanels[v.Category].Registered = true
				end
			end

			Stop = true -- this prevents from the loop creating multiple same items
		end
	end
end

concommand.Add("dec_buymenu", DEC_BuyMenu)

local Calibers = {
	["9x19MM"] = {cal = "9x19MM firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	["5.7x28MM"] = {cal = "5.7x28MM firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	[".45 ACP"] = {cal = ".45 ACP firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	[".50 AE"] = {cal = ".50 AE firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	["7.62x39MM"] = {cal = "7.62x39MM firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	["7.62x51MM"] = {cal = "7.62x51MM firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	["5.56x45MM"] = {cal = "5.56x45MM firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	[".338 Lapua Magnum"] = {cal = ".338 Lapua Magnum firearm", func = function(text)
		return "It looks like this person was killed by\na firearm chambered in the\n" .. text .. " caliber."
	end},

	["Clumsiness"] = {cal = "Clumsiness", func = function(text)
		return "It seems that this person fell to his death.\nThe cause is not known, it could be\nanything."
	end},

	["Steel"] = {cal = "Steel knife", func = function(text)
		return "It looks like this person was killed by\na knife."
	end},

	["Poison dart"] = {cal = "Dart", func = function(text)
		return "It looks like this person was killed by\nsome kind of a dart.\nMost likely a poison dart."
	end}}

local function DEC_ExamineDeadBody(um)
	body = um:ReadEntity()
	cal = um:ReadString()
	time = um:ReadShort()
	role = um:ReadShort()
	ply = LocalPlayer()

	addy = 0

	if ply.TeamText == "Agent" then
		addy = 30
	end

	MainFrame2 = create("DFrame2")
	MainFrame2:SetSize(250, 360 + addy)
	MainFrame2:Center()
	MainFrame2:SetTitle("Examining " .. body:Nick() .. "'s body.")
	MainFrame2:SetVisible(true)
	MainFrame2:SetDraggable(false)
	MainFrame2:ShowCloseButton(true)
	MainFrame2:MakePopup()

	MainFrame2.Alpha = 200

	KillCaliber = create("DLabel", MainFrame2)
	print(cal)
	KillCaliber:SetText("Kill cause - " .. (Calibers[cal] and Calibers[cal].cal or "The world"))
	KillCaliber:SetPos(10, 30)
	KillCaliber:SetFont("TabLarge")
	KillCaliber:SizeToContents()

	KillCaliber = create("DImage", MainFrame2)
	KillCaliber:SetPos(10, 50)
	KillCaliber:SetImage("deception/icon_killcaliber_2")
	KillCaliber:SetSize(48, 48)

	KillCaliber = create("DLabel", MainFrame2)
	KillCaliber:SetText(Calibers[cal] and Calibers[cal].func(cal) or "Unknown")
	KillCaliber:SetPos(10, 100)
	KillCaliber:SetFont("TabLarge")
	KillCaliber:SizeToContents()

	CT = CurTime()
	min = CT - time <= 59 and "minute" or "minutes"

	KillTime = create("DLabel", MainFrame2)
	KillTime:SetText("Kill time - " .. math.ceil((CT - time) / 60) .. " " .. min .. " ago")
	KillTime:SetPos(10, 160)
	KillTime:SetFont("TabLarge")
	KillTime:SizeToContents()

	KillTime = create("DImage", MainFrame2)
	KillTime:SetPos(10, 180)
	KillTime:SetImage("deception/icon_killtime_2")
	KillTime:SetSize(48, 48)

	text = nil

	if math.ceil((CT - time)/ 60) >= 3 then
		text = "The body is cold, the kill was most likely\nperformed quite some time ago."
	else
		text = "The body is fresh, the kill was most likely\nperformed not long ago."
	end

	KillTime = create("DLabel", MainFrame2)
	KillTime:SetText(text)
	KillTime:SetPos(10, 230)
	KillTime:SetFont("TabLarge")
	KillTime:SizeToContents()

	-- role

	rolet = nil

	if role == 1 then
		rolet = "Civilian"
	elseif role == 2 then
		rolet = "Agent"
	elseif role == 3 then
		rolet = "VIP"
	end

	KillTime = create("DLabel", MainFrame2)
	KillTime:SetText("Player role - " .. rolet)
	KillTime:SetPos(10, 280)
	KillTime:SetFont("TabLarge")
	KillTime:SizeToContents()

	KillTime = create("DImage", MainFrame2)
	KillTime:SetPos(10, 300)
	KillTime:SetImage("deception/icon_playerrole")
	KillTime:SetSize(48, 48)

	-- report dead body button for agents

	if ply.TeamText == "Agent" then
		RepButt = create("DButton", MainFrame2) -- almost RAPEBUTT, AMIRITE?! LOLOLO XDXDXDD LE EPIC WIN FOR LE REDDIT!!! no, fuck you
		RepButt:SetPos(10, 360)
		RepButt:SetSize(100, 18)
		RepButt:SetText("Report body")
		RepButt.DoClick = function()
			RunConsoleCommand("dec_reportdeadbody")
		end
	end
end

usermessage.Hook("EXAMINE_DEADBODY", DEC_ExamineDeadBody)

local function DEC_HelpMenu()
	MainFrame3 = create("DFrame2")
	MainFrame3:SetSize(400, 360)
	MainFrame3:Center()
	MainFrame3:SetTitle("Help menu.")
	MainFrame3:SetVisible(true)
	MainFrame3:SetDraggable(false)
	MainFrame3:ShowCloseButton(true)
	MainFrame3:MakePopup()

	MainFrame3.Alpha = 150

	PSheet = create("DPropertySheet2", MainFrame3)
	PSheet:SetPos(10, 25)
	PSheet:SetSize(MainFrame3:GetWide() - 20, MainFrame3:GetTall() - 30)

	HelpPanel = create("DPanelList2", PSheet)
	HelpPanel:SetSize(380, 320)

	Help = create("DLabel", HelpPanel)
	Help:SetText("The basics.")
	Help:SetPos(0, 0)
	Help:SetFont("TabLarge")
	Help:SetTextColor(Color(0, 200, 80, 255))
	Help:SizeToContents()

	Help = create("DLabel", HelpPanel)
	Help:SetText([[Out of all players, 3 are chosen, 2 of which are Agents A and B.
The third person is the VIP, everyone else remains a Civilian.
To win the round, the Agents have to kill each other AND the VIP.
The VIP has to survive the round OR eliminate the Agents to win.
The Civilians are sworn protect the VIP.]])

	Help:SetFont("DefaultBoldShadow")
	Help:SetPos(0, 20)
	Help:SizeToContents()

	Help = create("DLabel", HelpPanel)
	Help:SetText("The gameplay.")
	Help:SetPos(0, 90)
	Help:SetFont("TabLarge")
	Help:SetTextColor(Color(0, 200, 80, 255))
	Help:SizeToContents()

	Help = create("DLabel", HelpPanel)
	Help:SetText([[Each player gains money over time to spend on items.
The player can access the buy menu by pressing F1 (gm_showhelp).
Some weapons are restricted to certain player classes only (Agent/VIP).
The Agents are able to pick-pocket players to find clues.
The VIP is able to show his VIP ID card to players, to prove that he
is the VIP and get protection from Civilians.
Every player is able to examine another player. To do that, the player
has to come up to another player and press his SPAWN MENU KEY (default - Q).
Agents are much more experienced in examining people and may notice
suspicious players much easier than regular Civilians will.
Each player also has a karma rating. Karma affects the amount of money
a player gets at the start of a round.
Having a karma that's too low will result in a 1 HOUR BAN.]])
	Help:SetFont("DefaultBoldShadow")
	Help:SetPos(0, 110)
	Help:SizeToContents()

	PSheet:AddSheet("Basics", HelpPanel, "gui/silkicons/user", false, false, "The basics of the gamemode.")

	HelpPanel = create("DPanelList2", PSheet)
	HelpPanel:SetSize(380, 320)

	Help = create("DLabel", HelpPanel)
	Help:SetText("Climbing.")
	Help:SetPos(0, 0)
	Help:SetFont("TabLarge")
	Help:SetTextColor(Color(0, 200, 80, 255))
	Help:SizeToContents()

	Help = create("DLabel", HelpPanel)
	Help:SetText([[Each player is able to grab onto ledges and pull himself up.
To do that, walk up to a wall, JUMP up, and hold down your JUMP KEY.
If it's possible, you will grab onto the ledge.
To pull yourself up, press your JUMP KEY again.
To let go of the ledge, press your CROUCH KEY.
You cannot grab onto the ledge while crouching.
Falling down too fast also disallows the ledge grab.
Holding onto a ledge or pulling self up drains stamina.]])

	Help:SetFont("DefaultBoldShadow")
	Help:SetPos(0, 20)
	Help:SizeToContents()

	Help = create("DLabel", HelpPanel)
	Help:SetText("Stamina.")
	Help:SetPos(0, 130)
	Help:SetFont("TabLarge")
	Help:SetTextColor(Color(0, 200, 80, 255))
	Help:SizeToContents()

	Help = create("DLabel", HelpPanel)
	Help:SetText([[Each player also has limited stamina.
Actions such as sprinting, jumping and climbing drain stamina.
When stamina is too low, the player will not be able to sprint.
The indication of stamina is purely visual and sound based.
When your stamina is low enough, you will hear a heartbeat sound
and the corners of your vision will become blurry and darkened.
Walking or standing still restores stamina.
Getting injured also drains stamina.
Your stamina will also drain faster if you have lots of equipment
(weapons, armor) on you.]])

	Help:SetFont("DefaultBoldShadow")
	Help:SetPos(0, 150)
	Help:SizeToContents()

	PSheet:AddSheet("Additional", HelpPanel, "deception/adv_help", false, false, "Additional info for those that don't get tired from reading.")

	HelpPanel = create("DPanelList2", PSheet)
	HelpPanel:SetSize(380, 320)

	mblur = GetConVarNumber("mat_motion_blur_enabled")

	CBox = create("DCheckBoxLabel", HelpPanel)
	CBox:SetText("Motion Blur")
	CBox:SetPos(0, 0)
	CBox:SetConVar(mblur)
	CBox:SetValue(mblur)
	CBox:SizeToContents()

	stamblur = GetConVarNumber("dec_staminablur")

	CBox = create("DCheckBoxLabel", HelpPanel)
	CBox:SetText("Stamina Blur (Requires Motion Blur!)")
	CBox:SetPos(0, 20)
	CBox:SetConVar(stamblur)
	CBox:SetValue(stamblur)
	CBox:SizeToContents()

	stambeat = GetConVarNumber("dec_staminabeat")

	CBox = create("DCheckBoxLabel", HelpPanel)
	CBox:SetText("Stamina Heartbeat")
	CBox:SetPos(0, 40)
	CBox:SetConVar(stambeat)
	CBox:SetValue(stambeat)
	CBox:SizeToContents()

	PSheet:AddSheet("Options", HelpPanel, "gui/silkicons/wrench", false, false, "Disable or enable some stuff to your own liking.")
end

concommand.Add("dec_helpmenu", DEC_HelpMenu)
